src/gmail.js

```diff
@@ -2561,6 +2567,31 @@ var Gmail = function(localJQuery) {
             return flag_value === "0" || flag_value === undefined;
         }
     };
+    
+	//To know button labels setting is set to text (or icon)
+    api.check.is_buttensLabel_text = function() {
+        if( api.check.is_new_data_layer() ) {
+            var flag = 0;
+            
+            if(document.querySelector("div[gh='tm'] div[class='Bn']") !== null) {
+                flag = 1;
+            }
+            
+            return flag;
+        } else {	//To handle classic gmail UI           	
+            var flag_name = "bx_sabi";
+            var flag_value = undefined;
+            var array_with_flag = api.tracker.globals[17][4][1];
+            for (var i = 0; i < array_with_flag.length; i++) {
+                var current = array_with_flag[i];
+                if (current[0] === flag_name) {
+                    flag_value = current[1];
+                    break;
+                }
+            }
```

Comment:
Right now its quite a few variables being thrown around making it unclear what the intent of the code is.

Could you extract this into a function of its own (`get_find_value_by_key` or something)? I think that would make the rest of the code much clearer.